src/views/catalog/CreateFromInstanceTypes/utils/utils.ts

```diff
@@ -8,11 +8,28 @@ import { DEFAULT_NAMESPACE } from '@kubevirt-utils/constants/constants';
 import { ALL_NAMESPACES_SESSION_KEY } from '@kubevirt-utils/hooks/constants';
 import { getRandomChars } from '@kubevirt-utils/utils/utils';
 
-import { DEFAULT_INSTANCETYPE_LABEL, DEFAULT_PREFERENCE_LABEL } from './constants';
+import { InstanceTypeSize } from '../components/SelectInstanceTypeSection/utils/types';
+import { categoryNamePrefixMatcher } from '../components/SelectInstanceTypeSection/utils/utils';
+
+import {
+  DEFAULT_INSTANCETYPE_LABEL,
+  DEFAULT_PREFERENCE_LABEL,
+  InstanceTypeState,
+} from './constants';
 
 const generateCloudInitPassword = () =>
   `${getRandomChars(4)}-${getRandomChars(4)}-${getRandomChars(4)}`;
 
+export const getInstanceTypeState = (defaultInstanceTypeName: string): InstanceTypeState => {
+  const [prefix, size] = defaultInstanceTypeName?.split('.');
```

Comment:
```suggestion
  const [prefix, size]: [string, InstanceTypeSize] = defaultInstanceTypeName?.split('.');
```

can work?

Reply:
then u can remove the as below

Reply:
will change it in a follow-up PR :) good suggestion

Reply:
@metalice adding your suggestion is not allowed from TS as string[] cannot be converted, giving following error:
```
Type 'string[]' is not assignable to type '[string, InstanceTypeSize]'.
  Target requires 2 element(s) but source may have fewer. 
  ```